qiskit/aqua/algorithms/single_sample/hhl/hhl.py

```diff
@@ -0,0 +1,328 @@
+        self._eigs = eigs
+        self._init_state = init_state
+        self._reciprocal = reciprocal
+        self._num_q = num_q
+        self._num_a = num_a
+        self._circuit = None
+        self._io_register = None
+        self._eigenvalue_register = None
+        self._ancilla_register = None
+        self._success_bit = None
+        self._ret = {}
+
+    @classmethod
+    def init_params(cls, params, algo_input):
+        """Initialize via parameters dictionary and algorithm input instance
+
+        Args:
+            params: parameters dictionary
+            algo_input: LinearSystemInput instance
+        """
+        if algo_input is None:
+            raise AquaError("LinearSystemInput instance is required.")
+
+        matrix = algo_input.matrix
+        vector = algo_input.vector
+        if not isinstance(matrix, np.ndarray):
+            matrix = np.array(matrix)
+        if not isinstance(vector, np.ndarray):
+            vector = np.array(vector)
+
+        # extend vector and matrix for nonhermitian/non 2**n size matrices
+        if np.log2(matrix.shape[0]) % 1 != 0:
+            next_higher = int(np.ceil(np.log2(matrix.shape[0])))
+            new_matrix = np.identity(2**next_higher)
+            new_matrix = np.array(new_matrix, dtype=complex)
+            new_matrix[:matrix.shape[0], :matrix.shape[0]] = matrix[:, :]
+            matrix = new_matrix
+            new_vector = np.ones((1, 2**next_higher))
+            new_vector[0, :vector.shape[0]] = vector
+            vector = new_vector.reshape(np.shape(new_vector)[1])
+        if matrix.shape[0] != len(vector):
+            raise ValueError("Input vector dimension does not match input "
+                             "matrix dimension!")
+
+        # Initialize eigenvalue finding module
+        eigs_params = params.get(Pluggable.SECTION_KEY_EIGS)
+        eigs = get_pluggable_class(PluggableType.EIGENVALUES,
+                                   eigs_params['name']).init_params(params, matrix)
+        num_q, num_a = eigs.get_register_sizes()
+
+        # Initialize initial state module
+        tmpvec = vector
+        init_state_params = params.get(Pluggable.SECTION_KEY_INITIAL_STATE)
+        init_state_params["num_qubits"] = num_q
+        init_state_params["state_vector"] = tmpvec
+        init_state = get_pluggable_class(PluggableType.INITIAL_STATE,
+                                         init_state_params['name']).init_params(params)
+
+        # Initialize reciprocal rotation module
+        reciprocal_params = params.get(Pluggable.SECTION_KEY_RECIPROCAL)
+        reciprocal_params["negative_evals"] = eigs._negative_evals
+        reciprocal_params["evo_time"] = eigs._evo_time
+        reci = get_pluggable_class(PluggableType.RECIPROCAL,
+                                   reciprocal_params['name']).init_params(params)
+
+        return cls(matrix, vector, eigs, init_state, reci, num_q, num_a)
+
+    def construct_circuit(self):
```

Comment:
could we add an argument `meansurement=False` instead of directly using `quantum_instance` in the `construct_circuit` method? In this way, users can get the circuit without running it. (or assign qunatum_instance by themselves.) , then, you just need to update line 315 accordingly.